net/netbird/src/opnsense/mvc/app/controllers/OPNsense/netbird/Api/ServiceController.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+/*
+ * Copyright (C) 2025 Ralph Moser, PJ Monitoring GmbH
+ * Copyright (C) 2025 squared GmbH
+ * Copyright (C) 2025 Christopher Linn, BackendMedia IT-Services GmbH
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\netbird\Api;
+
+use OPNsense\Base\ApiMutableServiceControllerBase;
+use OPNsense\Core\Backend;
+use OPNsense\Core\Config;
+use OPNsense\netbird\Initial;
+use OPNsense\netbird\Netbird;
+
+
+/**
+ * Class ServiceController
+ * @package OPNsense\netbird
+ */
+class ServiceController extends ApiMutableServiceControllerBase
+{
+    const NETBIRD_CONFIG_JSON = '/usr/local/etc/netbird/config.json';
```

Comment:
we're working on privilege separation and writing the config from the controller will eventually stop working. we use configd templating and backend scripts to achieve this

Reply:
yes, that would be preferable